mono/eglib/glib.h

```diff
@@ -329,7 +571,13 @@ typedef void     (*GFreeFunc)      (gpointer       data);
  */
 typedef struct _GSList GSList;
 struct _GSList {
+#ifdef __cplusplus
+	// Allow for conversion of list->data to anything, in context.
+	// This saves adding casts to turn valid C into valid C++.
+	g_cast_t<gpointer> data;
```

Comment:
This makes `_GSList` not a POD anymore.  Is this where the non-`explicit` default constructor for `g_cast_t<>` is used?

Reply:
This is not where it is used. There is a local variable out in the tree. I tried other fixed.

Reply:
I'm confused/surprised how there can be a local variable of `g_cast_t<>` type. It's a brand new type.

Reply:
Local of ‘GSList’ or ‘GList’.

Reply:
POD-ness should not affect layout. Perhaps ABI of pass/return-by-value, which should be rare/zero. We can cast in list uses instead. It isn’t as much as allocators. Also C/C++ should be pretty rare, that traffic in glib, if we convert utils/metadata/sgen/mini, as I have.